compiler/rustc_metadata/src/rmeta/decoder/cstore_impl.rs

```diff
@@ -44,18 +44,33 @@ macro_rules! provide {
                 let ($def_id, $other) = def_id_arg.into_args();
                 assert!(!$def_id.is_local());
 
-                let $cdata = CStore::from_tcx($tcx).get_crate_data($def_id.krate);
-
                 if $tcx.dep_graph.is_fully_enabled() {
```

Comment:
You could add a `&& stringify!($name) != "crate_hash"` test and get away without creating a dedicated `crate_hash` function.

Reply:
Beautiful, thank you!